src/main/java/org/cactoos/collection/Immutable.java

```diff
@@ -70,7 +71,7 @@ public Immutable(final X... src) {
      * @param src Source iterable
      */
     public Immutable(final Iterable<X> src) {
-        this(new CollectionOf<>(src));
+        this(new ListOf<X>(src));
```

Comment:
@fanifieiev I think here it would make more sense to just remove this constructor because it makes no sense to decorate an `Iterable` with a `Collection` decorator!

Reply:
@victornoel What is the sense of  this class then?

Reply:
@fanifieiev it's a decorator for `Collection`, not for `Iterable`. If you want to decorate an `Iterable`, you can use the decorators in the `collection` package.

Reply:
@fanifieiev also I share your view on the non sense of most of the classes in the collection package, but `Immutable` does make sense actually, it decorates a `Collection` so that it behaves as `Immutable`.

Reply:
> @fanifieiev it's a decorator for `Collection`, not for `Iterable`. If you want to decorate an `Iterable`, you can use the decorators in the `collection` package.

@victornoel Agree, but let's take for example [org.cactoos.collection.HeadOf](https://github.com/yegor256/cactoos/blob/master/src/main/java/org/cactoos/collection/HeadOf.java). It does accept `Iterable`, but probably should accept `Collection`.
The same is for other decorators in the collection package

Reply:
@fanifieiev 
> Agree, but let's take for example org.cactoos.collection.HeadOf. It does accept Iterable, but probably should accept Collection.

I believe `collection.HeadOf` is useless, because there is already `iterable.HeadOf` that does the same thing.

> I think we need to narrow the number of constructors in the decorators.

Exactly my point: if it makes sense for a decorator of collections/iterable to have an `Iterable` as argument, then it means it belongs to `iterable` package, e.g., `HeadOf`. If it doesn't make sense then it's good it is in `collection` package, e.g., `Immutable` here.